api/src/main/java/org/openmrs/module/fhir2/api/dao/impl/FhirObservationDaoImpl.java

```diff
@@ -44,17 +50,62 @@ public Obs getObsByUuid(String uuid) {
 	
 	@Override
 	public Collection<Obs> searchForObservations(ReferenceAndListParam encounterReference,
-	        ReferenceAndListParam patientReference, TokenAndListParam code, SortSpec sort) {
+	        ReferenceAndListParam patientReference, ReferenceParam hasMemberReference, TokenAndListParam valueConcept,
+	        DateRangeParam valueDateParam, QuantityAndListParam valueQuantityParam, StringAndListParam valueStringParam,
+	        DateRangeParam date, TokenAndListParam code, SortSpec sort) {
+		
 		Criteria criteria = sessionFactory.getCurrentSession().createCriteria(Obs.class);
 		
 		handleEncounterReference("e", encounterReference).ifPresent(c -> criteria.createAlias("encounter", "e").add(c));
 		handlePatientReference(criteria, patientReference, "person");
+		handleHasMemberReference(criteria, hasMemberReference);
+		handleValueCodedConcept(criteria, valueConcept);
+		handleDateRange("valueDatetime", valueDateParam);
+		
+		handleValueStringParam("valueText", valueStringParam).ifPresent(criteria::add);
+		handleValueQuantityParam("valueNumeric", valueQuantityParam).ifPresent(criteria::add);
+		handleDateRange("obsDatetime", date);
 		handleCodedConcept(criteria, code);
 		handleSort(criteria, sort);
 		
 		return criteria.list();
 	}
 	
+	protected void handleHasMemberReference(Criteria criteria, ReferenceParam hasMemberReference) {
+		if (hasMemberReference != null) {
+			criteria.createAlias("groupMembers", "gm");
+			
+			if (hasMemberReference.getChain() != null) {
+				switch (hasMemberReference.getChain()) {
+					case Observation.SP_CODE:
+						TokenAndListParam code = new TokenAndListParam()
+						        .addAnd(new TokenParam().setValue(hasMemberReference.getValue()));
+						criteria.createAlias("gm.concept", "c");
+						findMatchingConcepts("c", code, criteria).ifPresent(criteria::add);
+						break;
+					case "":
+						criteria.add(eq("gm.uuid", hasMemberReference.getIdPart()));
+						break;
+				}
+			}
+		}
+	}
+	
+	protected Optional<Criterion> handleValueQuantityParam(@NotNull String propertyName,
```

Comment:
I'd rename this `handleQuantity()` and move it to BaseDaoImpl. I'd also rename the parameter `quantityAndListParam`.